lib/smart_answer/calculators/holiday_entitlement.rb

```diff
@@ -37,6 +41,48 @@ def initialize(working_days_per_week: 0,
       @days_per_shift_pattern = BigDecimal(days_per_shift_pattern, 10)
     end
```

Comment:
Looking at the flow where this calculator is used, passing attributes in to the calculator on initialization is only used in tests ..... and as you've removed most of those assignments from the test, I think the `initialize` method can be significantly simplified.

It seems a shame to do all that work in the tests to remove the assignment of attributes there, to then not reap the benefit by simplifying the code here.

Reply:
Thanks @reggieb - this is much cleaner